src/Objects/_PVector.js

```diff
@@ -0,0 +1,283 @@
+    }
+    return target? target.set(v.x*n.x, v.y*n.y, v.z*n.z)
+                 : new _PVector(v.x*n.x, v.y*n.y, v.z*n.z);
+  };
+
+  _PVector.div = function(v, n, target) {
+    if (typeof n === 'number') {
+      return target? target.set(v.x/n, v.y/n, v.z/n)
+                   : new _PVector(v.x/n, v.y/n, v.z/n);
+    }
+    return target? target.set(v.x/n.x, v.y/n.y, v.z/n.z)
+                 : new _PVector(v.x/n.x, v.y/n.y, v.z/n.z);
+  };
+
+  _PVector.prototype = {
+    copy: function() {
+      return new _PVector(this.x, this.y, this.z);
+    },
+    get: function(target) {
+      if (!arguments.length) { return this.copy(); } // @Deprecated
+      if (typeof target !== 'object') { return this.array(); }
+      target[0] = this.x;
+      target[1] = this.y;
+      target[2] = this.z;
+      return target;
+    },
+    set: function(v, y, z) {
+      const len = arguments.length;
+      if (len > 1) {
+        this.x = v;
+        this.y = y;
+        if (len > 2) { this.z = z; }
+      } else if (len === 1) {
+        this.set(v.x || v[0] || 0,
+                 v.y || v[1] || 0,
+                 v.z || v[2] || 0);
+      } else { argsErr('set', len, 1); }
+      return this;
+    },
+    array: function() {
+      return [this.x, this.y, this.z];
+    },
+    normalize: function(target) {
+      const m = this.mag(),
+            canDivide = m !== 0 && m !== 1;
+      if (!arguments.length) { return canDivide? this.div(m) : this; }
+      return canDivide? _PVector.div(this, m, target)
+                      : target? target.set(this) : this.copy();
+    },
+    limit: function(max) {
+      return this.magSq() > max*max? this.normalize().mult(max) : this;
+    },
+    heading: function() {
+      //return -Math.atan2(-this.y, this.x);
+      return Math.atan2(this.y, this.x);
+    },
+    heading2D: null, // @Deprecated
+    mag: function() {
+      return Math.sqrt(this.magSq());
+    },
+    magSq: function() {
+      return sq(this.x) + sq(this.y) + sq(this.z);
+    },
+    setMag: function(target, length) {
+      const len = arguments.length;
+      return len === 1? this.normalize().mult(target) :
+             len > 1? this.normalize(target).mult(length) :
+             argsErr('setMag', len, 1);
+    },
+    rotate: function(angle) {
+      const prev_x = this.x,
+            c = Math.cos(angle),
+            s = Math.sin(angle);
+      this.x = c*this.x - s*this.y;
+      this.y = s*prev_x + c*this.y;
+      return this;
+    },
+    dist: function(v1, v2) {
+      return v2? _PVector.dist(v1, v2) : _PVector.dist(this, v1);
+    },
+    distSq: function(v1, v2) {
+      return v2? _PVector.distSq(v1, v2) : _PVector.distSq(this, v1);
+    },
+    dot: function(v, y, z) {
+      const len = arguments.length;
+      return len === 1? _PVector.dot(this, v) :
+             len === 2? _PVector.dot(v, y) :
+             len > 2? this.x*v + this.y*y + this.z*z :
+             argsErr('dot', len, 1);
```

Comment:
Try not to optimize prematurely. For any Open Source project, the development cycle is pretty much 1) make it work, 2) make it work faster. And if the code is small enough, things like JIT typically mean that you typically don't get to (2). As the perf test lower down shows, using ternaries might be faster in compiled languages, but in JS it almost never matters because the code's compiler-optimised just before it's invoked in the browser anyway.

(it's something I had to learn when I went from being a Java developer to being a JavaScript developer, too. Things that are obvious, and easy, optimizations in Java, are often not optimizations *at all* in JavaScript)